src/blocks/content-helper/components/post-list.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { Spinner } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ContentHelperProvider from '../content-helper-provider';
+import PostCard from './post-card';
+import { SuggestedPost } from '../models/suggested-post';
+
+const FETCH_RETRIES = 3;
+
+function PostList() {
+	const [ loading, setLoading ] = useState<boolean>( true );
+	const [ error, setError ] = useState<string>( null );
+	const [ message, setMessage ] = useState<string>( null );
+	const [ posts, setPosts ] = useState<SuggestedPost[]>( [] );
+
+	const fetchPosts = async ( retries: number ) => {
+		ContentHelperProvider.getTopPosts()
+			.then( ( result ) => {
+				setPosts( result.posts );
+				setMessage( result.message );
+				setLoading( false );
+			} )
+			.catch( async ( err: string ) => {
```

Comment:
Is `err` a `string`? You might want to access `err.message`

Reply:
This seems correct to me.

Reply:
Thanks for the info, this is something to keep in mind for me.

In this case, [fetchData()](https://github.com/Parsely/wp-parsely/blob/3207db79d69e06711e1e4fc0c1eba9a61e802f34/src/blocks/content-helper/content-helper-provider.ts#L62-L79) returns an array of results or a string error.

Maybe not an ideal solution (?). For the time being I propose to leave as is and work on it in the future if you think it may be worth it?

Reply:
Upon further examination what I said above is partly true, because there's a case where an error is returned.